src/dataset.jl

```diff
@@ -255,8 +257,36 @@ function writelayers(
     dataset::AbstractDataset,
     filename::AbstractString;
     driver::Driver = getdriver(dataset),
-    options = [""],
-    layer_options = [""],
+    options = String[],
+    layers = 0:(nlayer(dataset)-1),
+    layer_options = String[],
+    chunksize::Integer = 20000,
+    use_gdal_copy::Bool = true,
+)
+    create(filename; driver = driver, options = options) do target
+        return copylayers!(
+            dataset,
+            target;
+            layers = layers,
+            layer_options = layer_options,
+            chunksize = chunksize,
+            use_gdal_copy = use_gdal_copy,
+        )
+    end
+    return nothing
+end
+
+"""
+    copylayers(source_dataset, target_dataset; kwargs...)
+
+Copies a (sub)set of layers from source to target dataset. The keyword arguments are the same as for `writelayers`.
+Note that eventual layer options have to conform with the target dataset driver.
+"""
+function copylayers!(
+    source_dataset::AbstractDataset,
+    target_dataset::AbstractDataset;
+    layers = 0:(nlayer(source_dataset)-1),
```

Comment:
Can we rename it from `layers` to `source_layer_indices` (or `source_layer_index` if it's `copylayer!`)?